circle.yml

```diff
@@ -1,22 +1,51 @@
+general:
+  # Artifacts that we want to collect from the build such as test results and logs
+  artifacts:
+    - "target/universal"
+    - "target/specs2-reports"
+    - "target/test-reports"
+    - "logs"
+
 machine:
   java:
     version: oraclejdk8
+  services:
+    - docker
+
 dependencies:
   pre:
-    - wget -q https://dl.bintray.com/sbt/debian/sbt-0.13.7.deb
-    - sudo dpkg -i sbt-0.13.7.deb
-    - wget -q http://downloads.typesafe.com/scala/2.11.7/scala-2.11.7.deb
-    - sudo dpkg -i scala-2.11.7.deb
+    - wget -q https://dl.bintray.com/sbt/debian/sbt-0.13.12.deb
+    - sudo dpkg -i sbt-0.13.12.deb
+  # Compile here so that the dependencies can be cached between builds and to make sure the code
+  # actually compiles before we do a bunch of other work.
   override:
-    - sbt update
-  cache_directories:
-    - ~/.ivy2
-    - ~/.sbt
-    - ~/.m2
+    - sbt test:compile
+
+# Use this section to setup tests external dependencies, not just databases.
+database:
+  override:
+    - docker login -e $DOCKER_EMAIL -u $DOCKER_USER -p $DOCKER_PASS
+
 test:
   override:
-    - sbt scalastyle
-    - sbt test
+    - sbt test scalastyle
   post:
-    - sbt universal:package-zip-tarball
-    - cp target/universal/*.tgz $CIRCLE_ARTIFACTS
+    - mkdir -p $CIRCLE_TEST_REPORTS/junit/
+    - cp target/test-reports/*.xml $CIRCLE_TEST_REPORTS/junit/
+
+deployment:
+  # Only package and build a docker image if the tests pass.
+  hub:
+    branch: /^(?!master).*/
+    commands:
+      - sbt universal:package-zip-tarball
+      - docker build -t keen/capillary:$CIRCLE_BUILD_NUM .
+      - docker tag -f keen/capillary:$CIRCLE_BUILD_NUM keen/capillary:$CIRCLE_BRANCH
+      - docker push keen/capillary
+  master:
+    branch: master
+    commands:
+      - sbt universal:package-zip-tarball
+      - docker build -t keen/capillary:$CIRCLE_BUILD_NUM .
+      - docker tag -f keen/capillary:$CIRCLE_BUILD_NUM keen/capillary:latest
```

Comment:
I agree about the not tagging every branch latest, which we weren’t, but i think it might be better to just not tag anything latest and force people to be explicit